app/models/machine.rb

```diff
@@ -1,5 +1,11 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
   belongs_to :owner
+
+  def avg_price
+    snacks.empty? ? ("There are no snacks.") : ('%.2f' % snacks.average(:price))
+  end
```

Comment:
the first test did _not_ like avg_price being called on a machine with no snacks. had to add logic here.

Reply:
i _did_ find `number_to_currency` but tried it and syntax errored. i abandoned it in favor of "i got something else to work first" but will keep it in mind for future.

Reply:
number_to_currency will go in your view since it is an action view helper. it works like this... 

```ruby
<%= number_to_currency(snack.price) %>
```